lib/puppet/type/file/content_uri.rb

```diff
@@ -0,0 +1,42 @@
+require 'net/http'
+require 'uri'
+
+module Puppet
+  Puppet::Type.type(:file).newparam(:content_uri) do
+
+    desc <<-'EOT'
+      A unique identifier for file content, which must be a URI which points
+      to a file in a module. This parameter allows the content of a file resource
+      to remain consistent regardless of code updates on the master.
```

Comment:
Does the fileserver handle things that aren't in modules? Seems like we could specifically refer to module updates.